### ISSUE TITLE
issue on installing simpletransformers on Kaggle

### ISSUE THREAD

**Issue by kamastry** [ThilinaRajapakse/simpletransformers#1109]
HI
i can't import simpletrasnformers (last releases) on Kaggle in OFF LINE mode;
I uploaded .whl file as " new data" in my kernel, but if I try to !pip install unfortunately many dependencies fails, lacks...then it's impossible import it correctly; of course if i enable internet connection  all works fine, but in that kaggle competition internet is not allowed..

can someone help me about?
thanks

**Comment by ThilinaRajapakse** [comment]
I think there was a Kaggle post explaining how to do it. But, I've never tried it myself so I don't remember how it was done.

**Comment by caprone** [comment]
HI
I solved installing all dependences singly in that order; it is also advisable to install the following specified versions , (for someone's usefulness )


`
# ---- NLP packages installing
# ---- ---- all simpletransf dependeces:
!pip install ../input/seqeval-09/seqeval-0.0.9-py3-none-any.whl
!pip install ../input/tqdm48/tqdm-4.48.2-py2.py3-none-any.whl
!pip install ../input/hugging-face08/huggingface_hub-0.0.8-py3-none-any.whl
!pip install ../input/datasets150/xxhash-2.0.0-cp37-cp37m-manylinux1_x86_64.whl
!pip install ../input/datasets-16/datasets-1.6.0-py3-none-any.whl
!pip install ../input/base58/base58-2.1.0-py3-none-any.whl
!pip install ../input/astor22/astor-0.8.1-py2.py3-none-any.whl
!pip install ../input/watchdog/watchdog-2.0.3-py3-none-manylinux2014_x86_64.whl
!pip install ../input/pydeck/pydeck-0.6.2-py2.py3-none-any.whl
!pip install ../input/validators/validators-0.18.2-py3-none-any.whl
!pip install ../input/streamlit/streamlit-0.80.0-py2.py3-none-any.whl
!pip install ../input/simple-transfomrers-61/simpletransformers-0.61.4-py3-none-any.whl
!pip install ../input/fsspec20/fsspec-2021.4.0-py3-none-any.whl
`

**Comment by Nighthyst** [comment]
This solution works for me. I only had to the list of libraries just before the installation of xxhash:

```
!pip uninstall tqdm -y
!pip install --upgrade --no-build-isolation --no-index --find-links=file:///kaggle/input//tqdm-4.48.2-py2.py3-none-any.whl tqdm

```
The first line is to uninstall tqdm for the notebook and the second to install the right version. If you don't do that you will get this error while trying to install datasets:
ERROR: Could not find a version that satisfies the requirement tqdm<4.50.0,>=4.27 (from datasets)
ERROR: No matching distribution found for tqdm<4.50.0,>=4.27

The version of tqdm need to be greater or lower than 4.24 but not above 4.50.0. I guess the default version in kaggle's notebooks is not in this range.

Also for some reasons the syntax:
`!pip install /kaggle/input//tqdm-4.48.2-py2.py3-none-any.whl`
doesn't works well for me thus I had to go with:
`!pip install --upgrade --no-build-isolation --no-index --find-links=file:///kaggle/input/tqdm-4.48.2-py2.py3-none-any.whl tqdm`
